# Binary options put, call statistical research 

This Notebook analyse forex data from EURUSD pair M1 timeframe for the binary options market 
the data is from the duration of january 1st 2020 to december 31st 2020, the aim of this research is to find a pattern 
the placing trades in the binary options markert, in other to accurately predict where a market will close, from 
the previous candles, if it closes above the open price of the candle in question it is a call option, if it closes below it is a put option, when the open price and the close price is the same this is a no trade option which is statistically rare, so this won't be a focal point in this study

In [1]:
#import the relevant packages
import pandas as pd
import numpy as np
import seaborn as sns
sns.set(color_codes=True)

In [2]:
#read the data from a csv file
data = pd.read_csv("EURUSD_M1_202001020600_202012310000.csv", sep="\t")
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0       2020.01.02  06:00:00  1.12132  1.12133  1.12128  1.12131         14   
1       2020.01.02  06:01:00  1.12131  1.12133  1.12131  1.12132         16   
2       2020.01.02  06:02:00  1.12132  1.12133  1.12131  1.12132          9   
3       2020.01.02  06:03:00  1.12132  1.12134  1.12132  1.12132         19   
4       2020.01.02  06:04:00  1.12132  1.12133  1.12132  1.12132          3   
...            ...       ...      ...      ...      ...      ...        ...   
371711  2020.12.30  23:56:00  1.23003  1.23009  1.22994  1.23007         30   
371712  2020.12.30  23:57:00  1.23007  1.23012  1.23003  1.23003         46   
371713  2020.12.30  23:58:00  1.23004  1.23007  1.22997  1.22997         40   
371714  2020.12.30  23:59:00  1.22987  1.22987  1.22961  1.22961          7   
371715  2020.12.31  00:00:00  1.22963  1.22978  1.22963  1.22975          7   

        <VOL>  <SPREAD>  
0           0         3  
1           0         4  
2           0         3  
3           0         3  
4           0         4  
...       ...       ...  
371711      0         0  
371712      0         0  
371713      0         0  
371714      0        10  
371715      0        28  

[371716 rows x 9 columns]

In [3]:
#drop the <VOL>, column as it is not needed in this research
data.drop("<VOL>", inplace=True, axis=1)

In [4]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0       2020.01.02  06:00:00  1.12132  1.12133  1.12128  1.12131         14   
1       2020.01.02  06:01:00  1.12131  1.12133  1.12131  1.12132         16   
2       2020.01.02  06:02:00  1.12132  1.12133  1.12131  1.12132          9   
3       2020.01.02  06:03:00  1.12132  1.12134  1.12132  1.12132         19   
4       2020.01.02  06:04:00  1.12132  1.12133  1.12132  1.12132          3   
...            ...       ...      ...      ...      ...      ...        ...   
371711  2020.12.30  23:56:00  1.23003  1.23009  1.22994  1.23007         30   
371712  2020.12.30  23:57:00  1.23007  1.23012  1.23003  1.23003         46   
371713  2020.12.30  23:58:00  1.23004  1.23007  1.22997  1.22997         40   
371714  2020.12.30  23:59:00  1.22987  1.22987  1.22961  1.22961          7   
371715  2020.12.31  00:00:00  1.22963  1.22978  1.22963  1.22975          7   

        <SPREAD>  
0              3  
1              4  
2              3  
3              3  
4              4  
...          ...  
371711         0  
371712         0  
371713         0  
371714        10  
371715        28  

[371716 rows x 8 columns]

In [5]:
#set call when open price is less than close price
data["<UP>"] = data["<OPEN>"] < data["<CLOSE>"]

In [6]:
#set put when open price is greater than close price
data["<DOWN>"] = data["<OPEN>"] > data["<CLOSE>"]

In [7]:
#no trade when open price is equal to close price
data["<NO_MOVE>"] = data["<OPEN>"] == data["<CLOSE>"]

In [8]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0       2020.01.02  06:00:00  1.12132  1.12133  1.12128  1.12131         14   
1       2020.01.02  06:01:00  1.12131  1.12133  1.12131  1.12132         16   
2       2020.01.02  06:02:00  1.12132  1.12133  1.12131  1.12132          9   
3       2020.01.02  06:03:00  1.12132  1.12134  1.12132  1.12132         19   
4       2020.01.02  06:04:00  1.12132  1.12133  1.12132  1.12132          3   
...            ...       ...      ...      ...      ...      ...        ...   
371711  2020.12.30  23:56:00  1.23003  1.23009  1.22994  1.23007         30   
371712  2020.12.30  23:57:00  1.23007  1.23012  1.23003  1.23003         46   
371713  2020.12.30  23:58:00  1.23004  1.23007  1.22997  1.22997         40   
371714  2020.12.30  23:59:00  1.22987  1.22987  1.22961  1.22961          7   
371715  2020.12.31  00:00:00  1.22963  1.22978  1.22963  1.22975          7   

        <SPREAD>   <UP>  <DOWN>  <NO_MOVE>  
0              3  False    True      False  
1              4   True   False      False  
2              3  False   False       True  
3              3  False   False       True  
4              4  False   False       True  
...          ...    ...     ...        ...  
371711         0   True   False      False  
371712         0  False    True      False  
371713         0  False    True      False  
371714        10  False    True      False  
371715        28   True   False      False  

[371716 rows x 11 columns]

In [9]:
#convert these columns to type int so they appear as one hot varibles
data[["<UP>", "<DOWN>", "<NO_MOVE>"]] = data[["<UP>", "<DOWN>", "<NO_MOVE>"]].astype("int32")

In [10]:
data

<DATE>    <TIME>   <OPEN>   <HIGH>    <LOW>  <CLOSE>  <TICKVOL>  \
0       2020.01.02  06:00:00  1.12132  1.12133  1.12128  1.12131         14   
1       2020.01.02  06:01:00  1.12131  1.12133  1.12131  1.12132         16   
2       2020.01.02  06:02:00  1.12132  1.12133  1.12131  1.12132          9   
3       2020.01.02  06:03:00  1.12132  1.12134  1.12132  1.12132         19   
4       2020.01.02  06:04:00  1.12132  1.12133  1.12132  1.12132          3   
...            ...       ...      ...      ...      ...      ...        ...   
371711  2020.12.30  23:56:00  1.23003  1.23009  1.22994  1.23007         30   
371712  2020.12.30  23:57:00  1.23007  1.23012  1.23003  1.23003         46   
371713  2020.12.30  23:58:00  1.23004  1.23007  1.22997  1.22997         40   
371714  2020.12.30  23:59:00  1.22987  1.22987  1.22961  1.22961          7   
371715  2020.12.31  00:00:00  1.22963  1.22978  1.22963  1.22975          7   

        <SPREAD>  <UP>  <DOWN>  <NO_MOVE>  
0              3     0       1          0  
1              4     1       0          0  
2              3     0       0          1  
3              3     0       0          1  
4              4     0       0          1  
...          ...   ...     ...        ...  
371711         0     1       0          0  
371712         0     0       1          0  
371713         0     0       1          0  
371714        10     0       1          0  
371715        28     1       0          0  

[371716 rows x 11 columns]

In [11]:
open_close = (data["<OPEN>"] - data["<CLOSE>"]) * 100000
open_low = (data["<OPEN>"] - data["<LOW>"]) * 100000
open_high = (data["<OPEN>"] - data["<HIGH>"]) * 100000
close_low = (data["<CLOSE>"] - data["<LOW>"]) * 100000
close_high = (data["<CLOSE>"] - data["<HIGH>"]) * 100000
high_low = (data["<HIGH>"] - data["<LOW>"]) * 100000

In [12]:
#create a dataframe of the subtracted columns
move_df = pd.DataFrame({"open_close" : open_close, 
                        "open_low"   : open_low,
                        "open_high"  : open_high,
                        "close_low"  : close_low,
                        "close_high" : close_high,
                        "high_low"   : high_low, })
move_df

open_close  open_low  open_high  close_low  close_high  high_low
0              1.0       4.0       -1.0        3.0        -2.0       5.0
1             -1.0       0.0       -2.0        1.0        -1.0       2.0
2              0.0       1.0       -1.0        1.0        -1.0       2.0
3              0.0       0.0       -2.0        0.0        -2.0       2.0
4              0.0       0.0       -1.0        0.0        -1.0       1.0
...            ...       ...        ...        ...         ...       ...
371711        -4.0       9.0       -6.0       13.0        -2.0      15.0
371712         4.0       4.0       -5.0        0.0        -9.0       9.0
371713         7.0       7.0       -3.0        0.0       -10.0      10.0
371714        26.0      26.0        0.0        0.0       -26.0      26.0
371715       -12.0       0.0      -15.0       12.0        -3.0      15.0

[371716 rows x 6 columns]

### create a future dataframe to store the next data i.e the candle in question  

In [13]:
future = data.iloc[1:,8:11]

In [14]:
future.columns = ["up*", "down*", "no_move*"]

In [15]:
future.reset_index(inplace=True)
future.drop(["index"], axis = 1, inplace=True)

In [16]:
#concat move_df and future to create a new dataframe
dataset = pd.concat([move_df, future], axis = 1)
dataset

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
0              1.0       4.0       -1.0        3.0        -2.0       5.0  1.0   
1             -1.0       0.0       -2.0        1.0        -1.0       2.0  0.0   
2              0.0       1.0       -1.0        1.0        -1.0       2.0  0.0   
3              0.0       0.0       -2.0        0.0        -2.0       2.0  0.0   
4              0.0       0.0       -1.0        0.0        -1.0       1.0  0.0   
...            ...       ...        ...        ...         ...       ...  ...   
371711        -4.0       9.0       -6.0       13.0        -2.0      15.0  0.0   
371712         4.0       4.0       -5.0        0.0        -9.0       9.0  0.0   
371713         7.0       7.0       -3.0        0.0       -10.0      10.0  0.0   
371714        26.0      26.0        0.0        0.0       -26.0      26.0  1.0   
371715       -12.0       0.0      -15.0       12.0        -3.0      15.0  NaN   

        down*  no_move*  
0         0.0       0.0  
1         0.0       1.0  
2         0.0       1.0  
3         0.0       1.0  
4         1.0       0.0  
...       ...       ...  
371711    1.0       0.0  
371712    1.0       0.0  
371713    1.0       0.0  
371714    0.0       0.0  
371715    NaN       NaN  

[371716 rows x 9 columns]

In [17]:
dataset.dropna(inplace=True)

In [18]:
dataset

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
0              1.0       4.0       -1.0        3.0        -2.0       5.0  1.0   
1             -1.0       0.0       -2.0        1.0        -1.0       2.0  0.0   
2              0.0       1.0       -1.0        1.0        -1.0       2.0  0.0   
3              0.0       0.0       -2.0        0.0        -2.0       2.0  0.0   
4              0.0       0.0       -1.0        0.0        -1.0       1.0  0.0   
...            ...       ...        ...        ...         ...       ...  ...   
371710        -9.0       1.0      -17.0       10.0        -8.0      18.0  1.0   
371711        -4.0       9.0       -6.0       13.0        -2.0      15.0  0.0   
371712         4.0       4.0       -5.0        0.0        -9.0       9.0  0.0   
371713         7.0       7.0       -3.0        0.0       -10.0      10.0  0.0   
371714        26.0      26.0        0.0        0.0       -26.0      26.0  1.0   

        down*  no_move*  
0         0.0       0.0  
1         0.0       1.0  
2         0.0       1.0  
3         0.0       1.0  
4         1.0       0.0  
...       ...       ...  
371710    0.0       0.0  
371711    1.0       0.0  
371712    1.0       0.0  
371713    1.0       0.0  
371714    0.0       0.0  

[371715 rows x 9 columns]

In [19]:
#get the absolute value of all the data
absolute_data = abs(dataset)
absolute_data

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
0              1.0       4.0        1.0        3.0         2.0       5.0  1.0   
1              1.0       0.0        2.0        1.0         1.0       2.0  0.0   
2              0.0       1.0        1.0        1.0         1.0       2.0  0.0   
3              0.0       0.0        2.0        0.0         2.0       2.0  0.0   
4              0.0       0.0        1.0        0.0         1.0       1.0  0.0   
...            ...       ...        ...        ...         ...       ...  ...   
371710         9.0       1.0       17.0       10.0         8.0      18.0  1.0   
371711         4.0       9.0        6.0       13.0         2.0      15.0  0.0   
371712         4.0       4.0        5.0        0.0         9.0       9.0  0.0   
371713         7.0       7.0        3.0        0.0        10.0      10.0  0.0   
371714        26.0      26.0        0.0        0.0        26.0      26.0  1.0   

        down*  no_move*  
0         0.0       0.0  
1         0.0       1.0  
2         0.0       1.0  
3         0.0       1.0  
4         1.0       0.0  
...       ...       ...  
371710    0.0       0.0  
371711    1.0       0.0  
371712    1.0       0.0  
371713    1.0       0.0  
371714    0.0       0.0  

[371715 rows x 9 columns]

In [20]:
absolute_data.describe()

open_close       open_low      open_high      close_low  \
count  371715.000000  371715.000000  371715.000000  371715.000000   
mean        9.916250       8.820416       8.933659       8.858147   
std        12.367232      12.196695      12.375837      11.769463   
min         0.000000       0.000000       0.000000       0.000000   
25%         2.000000       1.000000       1.000000       2.000000   
50%         6.000000       5.000000       5.000000       5.000000   
75%        13.000000      12.000000      12.000000      12.000000   
max       665.000000     455.000000     797.000000     667.000000   

          close_high       high_low            up*          down*  \
count  371715.000000  371715.000000  371715.000000  371715.000000   
mean        8.895928      17.754075       0.465588       0.457485   
std        11.718535      16.297426       0.498815       0.498190   
min         0.000000       0.000000       0.000000       0.000000   
25%         2.000000       8.000000       0.000000       0.000000   
50%         5.000000      14.000000       0.000000       0.000000   
75%        12.000000      23.000000       1.000000       1.000000   
max       534.000000     809.000000       1.000000       1.000000   

            no_move*  
count  371715.000000  
mean        0.076927  
std         0.266476  
min         0.000000  
25%         0.000000  
50%         0.000000  
75%         0.000000  
max         1.000000

In [21]:
absolute_data.quantile([.95, .99, .995, .999])

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
0.950      32.000      31.0     30.000       30.0        30.0    46.000  1.0   
0.990      57.000      56.0     56.000       54.0        54.0    79.000  1.0   
0.995      71.000      70.0     70.000       67.0        68.0    96.000  1.0   
0.999     111.286     110.0    112.286      105.0       104.0   141.286  1.0   

       down*  no_move*  
0.950    1.0       1.0  
0.990    1.0       1.0  
0.995    1.0       1.0  
0.999    1.0       1.0

In [22]:
#display frequencies of up, down and no move
dataset[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         173066
0.0  1.0    0.0         170054
     0.0    1.0          28595
dtype: int64

# 95 percentile

### Anaylyse candles after a prior large drop in prices up to 3.2 pips in a minute open_close

In [23]:
big_drop_oc_95 = dataset[dataset["open_close"] > 32]
big_drop_oc_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
246           36.0      36.0        0.0        0.0       -36.0      36.0  0.0   
609           38.0      38.0       -3.0        0.0       -41.0      41.0  0.0   
764           47.0      52.0       -2.0        5.0       -49.0      54.0  1.0   
775           39.0      51.0       -4.0       12.0       -43.0      55.0  1.0   
845           37.0      41.0        0.0        4.0       -37.0      41.0  1.0   
...            ...       ...        ...        ...         ...       ...  ...   
371346        35.0      35.0      -24.0        0.0       -59.0      59.0  0.0   
371349        77.0      96.0       -4.0       19.0       -81.0     100.0  1.0   
371353        61.0      66.0      -10.0        5.0       -71.0      76.0  1.0   
371356        38.0      40.0       -2.0        2.0       -40.0      42.0  0.0   
371357        49.0      53.0      -22.0        4.0       -71.0      75.0  1.0   

        down*  no_move*  
246       1.0       0.0  
609       1.0       0.0  
764       0.0       0.0  
775       0.0       0.0  
845       0.0       0.0  
...       ...       ...  
371346    1.0       0.0  
371349    0.0       0.0  
371353    0.0       0.0  
371356    1.0       0.0  
371357    0.0       0.0  

[9347 rows x 9 columns]

In [24]:
#get the number of up and down movement in price after a big downward movement in price
big_drop_oc_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         4761
0.0  1.0    0.0         4384
     0.0    1.0          202
dtype: int64

### Result

From the information above we had 4761 upward movement in price and 4384 downward movement in price total <br/>
and 202 no movement in price <br/>
candles = 4761 + 4384 + 202 = 9347 <br/>
percentage up = (100 * 4761)/(9347) = 50.94% <br/>
percentage down = (100 * 4384)/(9347) = 46.90%  <br/>
percentage no move = (100 * 202)/(9347) = 2.16%   <br/>
so after a 3.2 pips rise in price for a one minute time frame of EURUSD there was a 50.94%  chance that the next candle would be bullish and 46.90% chance of a bearish candle and 2.16% chance of no movement

### Anaylyse candles after a prior large rise in prices up to 3.2 pips in a minute open_close

In [25]:
big_rise_oc_95 = dataset[dataset["open_close"] < -32]
big_rise_oc_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
421          -32.0       0.0      -33.0       32.0        -1.0      33.0  1.0   
694          -35.0       0.0      -46.0       35.0       -11.0      46.0  1.0   
716          -67.0       0.0      -74.0       67.0        -7.0      74.0  1.0   
718          -47.0       0.0      -47.0       47.0         0.0      47.0  1.0   
1641         -42.0       0.0      -43.0       42.0        -1.0      43.0  0.0   
...            ...       ...        ...        ...         ...       ...  ...   
371351       -41.0       0.0      -44.0       41.0        -3.0      44.0  1.0   
371352       -59.0       4.0      -63.0       63.0        -4.0      67.0  0.0   
371354       -52.0       0.0      -66.0       52.0       -14.0      66.0  1.0   
371378       -34.0       5.0      -34.0       39.0         0.0      39.0  0.0   
371595       -35.0       0.0      -35.0       35.0         0.0      35.0  1.0   

        down*  no_move*  
421       0.0       0.0  
694       0.0       0.0  
716       0.0       0.0  
718       0.0       0.0  
1641      1.0       0.0  
...       ...       ...  
371351    0.0       0.0  
371352    1.0       0.0  
371354    0.0       0.0  
371378    1.0       0.0  
371595    0.0       0.0  

[8943 rows x 9 columns]

In [26]:
big_rise_oc_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         4448
1.0  0.0    0.0         4304
0.0  0.0    1.0          191
dtype: int64

From the information above we had 4448 upward movement in price and 4304 downward movement in price total <br/>
and 191 no move that is open and close are equal
candles = 4448 + 4304 + 191 = 8943 <br/>
percentage up = (100 * 4448)/(8943) = 49.74% <br/>
percentage down = (100 * 4304)/(8943) = 48.13% <br/>
percentage no move = (100 * 191)/(8943) = 2.13% <br/>
so after a 3.2 pips rise in price for a one minute time frame of EURUSD there was a 49.74% chance that the next candle would be bullish and 48.13% chance of a bearish candle and 1.58% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 3.1 pips in a minute open_low

In [27]:
big_drop_ol_95 = dataset[dataset["open_low"] > 31]
big_drop_ol_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
246           36.0      36.0        0.0        0.0       -36.0      36.0  0.0   
570           25.0      34.0        0.0        9.0       -25.0      34.0  0.0   
583           18.0      36.0        0.0       18.0       -18.0      36.0  1.0   
605           21.0      32.0        0.0       11.0       -21.0      32.0  0.0   
609           38.0      38.0       -3.0        0.0       -41.0      41.0  0.0   
...            ...       ...        ...        ...         ...       ...  ...   
371357        49.0      53.0      -22.0        4.0       -71.0      75.0  1.0   
371406        21.0      32.0        0.0       11.0       -21.0      32.0  1.0   
371410        23.0      34.0       -8.0       11.0       -31.0      42.0  0.0   
371434        28.0      35.0       -7.0        7.0       -35.0      42.0  1.0   
371594        27.0      31.0        0.0        4.0       -27.0      31.0  1.0   

        down*  no_move*  
246       1.0       0.0  
570       1.0       0.0  
583       0.0       0.0  
605       1.0       0.0  
609       1.0       0.0  
...       ...       ...  
371357    0.0       0.0  
371406    0.0       0.0  
371410    1.0       0.0  
371434    0.0       0.0  
371594    0.0       0.0  

[18715 rows x 9 columns]

In [28]:
#get the number of up and down movement in price after a big downward movement in price
big_drop_ol_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         9413
0.0  1.0    0.0         8860
     0.0    1.0          442
dtype: int64

From the information above we had 9413 upward movement in price and 8860 downward movement in price total <br/>
and 442 no move that is open and close are equal
candles = 9413 + 8860 + 442 = 18715 <br/>
percentage up = (100 * 9413)/(18715) = 50.30% <br/>
percentage down = (100 * 8860)/(18715) = 47.34% <br/>
percentage no move = (100 * 442)/(18715) = 2.36% <br/>
so after a 3.1 pips drop in price for a one minute time frame of EURUSD there was a 49.74% chance that the next candle would be bullish and 48.13% chance of a bearish candle and 1.58% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 3.0 pips in a minute open_low

In [30]:
big_rise_oh_95 = dataset[dataset["open_high"] < -30]
big_rise_oh_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
421          -32.0       0.0      -33.0       32.0        -1.0      33.0  1.0   
482          -31.0       2.0      -32.0       33.0        -1.0      34.0  0.0   
694          -35.0       0.0      -46.0       35.0       -11.0      46.0  1.0   
715          -29.0       0.0      -38.0       29.0        -9.0      38.0  1.0   
716          -67.0       0.0      -74.0       67.0        -7.0      74.0  1.0   
...            ...       ...        ...        ...         ...       ...  ...   
371352       -59.0       4.0      -63.0       63.0        -4.0      67.0  0.0   
371354       -52.0       0.0      -66.0       52.0       -14.0      66.0  1.0   
371378       -34.0       5.0      -34.0       39.0         0.0      39.0  0.0   
371399       -31.0       5.0      -37.0       36.0        -6.0      42.0  0.0   
371595       -35.0       0.0      -35.0       35.0         0.0      35.0  1.0   

        down*  no_move*  
421       0.0       0.0  
482       1.0       0.0  
694       0.0       0.0  
715       0.0       0.0  
716       0.0       0.0  
...       ...       ...  
371352    1.0       0.0  
371354    0.0       0.0  
371378    1.0       0.0  
371399    1.0       0.0  
371595    0.0       0.0  

[18777 rows x 9 columns]

In [31]:
#get the number of up and down movement in price after a big upward movement in price
big_rise_oh_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         9152
0.0  1.0    0.0         9146
     0.0    1.0          479
dtype: int64

From the information above we had 9152 upward movement in price and 9146 downward movement in price total <br/>
and 479 no move that is open and close are equal
candles = 9152 + 9146 + 479 = 18777 <br/>
percentage up = (100 * 9152)/(18777) = 48.74% <br/>
percentage down = (100 * 9146)/(18777) = 48.71% <br/>
percentage no move = (100 * 479)/(18777) = 2.55% <br/>
so after a 3.1 pips drop in price for a one minute time frame of EURUSD there was a 48.74% chance that the next candle would be bullish and 48.71% chance of a bearish candle and 1.58% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 3.0 pips in a minute close_low

In [32]:
big_drop_cl_95 = dataset[dataset["close_low"] > 30]
big_drop_cl_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
421          -32.0       0.0      -33.0       32.0        -1.0      33.0  1.0   
482          -31.0       2.0      -32.0       33.0        -1.0      34.0  0.0   
694          -35.0       0.0      -46.0       35.0       -11.0      46.0  1.0   
716          -67.0       0.0      -74.0       67.0        -7.0      74.0  1.0   
718          -47.0       0.0      -47.0       47.0         0.0      47.0  1.0   
...            ...       ...        ...        ...         ...       ...  ...   
371354       -52.0       0.0      -66.0       52.0       -14.0      66.0  1.0   
371355       -13.0      25.0      -15.0       38.0        -2.0      40.0  0.0   
371378       -34.0       5.0      -34.0       39.0         0.0      39.0  0.0   
371399       -31.0       5.0      -37.0       36.0        -6.0      42.0  0.0   
371595       -35.0       0.0      -35.0       35.0         0.0      35.0  1.0   

        down*  no_move*  
421       0.0       0.0  
482       1.0       0.0  
694       0.0       0.0  
716       0.0       0.0  
718       0.0       0.0  
...       ...       ...  
371354    0.0       0.0  
371355    1.0       0.0  
371378    1.0       0.0  
371399    1.0       0.0  
371595    0.0       0.0  

[17672 rows x 9 columns]

In [33]:
big_drop_cl_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
0.0  1.0    0.0         8703
1.0  0.0    0.0         8537
0.0  0.0    1.0          432
dtype: int64

From the information above we had 8537 upward movement in price and 85 downward movement in price total <br/>
and 479 no move that is open and close are equal
candles = 8537 + 8703 + 432 = 17672<br/>
percentage up = (100 * 8537)/(17672) = 48.31% <br/>
percentage down = (100 * 8704)/(17672) = 49.35% <br/>
percentage no move = (100 * 432)/(17672) = 2.44% <br/>
so after a 3.0 pips drop in price for a one minute time frame of EURUSD there was a 48.31% chance that the next candle would be bullish and 49.35% chance of a bearish candle and 2.44% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 3.0 pips in a minute close_high

In [34]:
big_drop_ch_95 = dataset[dataset["close_high"] < -30]
big_drop_ch_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
246           36.0      36.0        0.0        0.0       -36.0      36.0  0.0   
609           38.0      38.0       -3.0        0.0       -41.0      41.0  0.0   
626           31.0      34.0        0.0        3.0       -31.0      34.0  0.0   
648           26.0      28.0       -8.0        2.0       -34.0      36.0  1.0   
682           27.0      27.0       -7.0        0.0       -34.0      34.0  1.0   
...            ...       ...        ...        ...         ...       ...  ...   
371356        38.0      40.0       -2.0        2.0       -40.0      42.0  0.0   
371357        49.0      53.0      -22.0        4.0       -71.0      75.0  1.0   
371377        19.0      19.0      -15.0        0.0       -34.0      34.0  1.0   
371410        23.0      34.0       -8.0       11.0       -31.0      42.0  0.0   
371434        28.0      35.0       -7.0        7.0       -35.0      42.0  1.0   

        down*  no_move*  
246       1.0       0.0  
609       1.0       0.0  
626       1.0       0.0  
648       0.0       0.0  
682       0.0       0.0  
...       ...       ...  
371356    1.0       0.0  
371357    0.0       0.0  
371377    0.0       0.0  
371410    1.0       0.0  
371434    0.0       0.0  

[18137 rows x 9 columns]

In [35]:
big_drop_ch_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         9145
0.0  1.0    0.0         8592
     0.0    1.0          400
dtype: int64

From the information above we had 9145 upward movement in price and 8592 downward movement in price total <br/>
and 400 no move that is open and close are equal
candles = 9145 + 8592 + 400 = 18137<br/>
percentage up = (100 * 9145)/(18137) = 50.42% <br/>
percentage down = (100 * 8592)/(18137) = 47.37% <br/>
percentage no move = (100 * 400)/(18137) = 2.21% <br/>
so after a 3.0 pips drop in price for a one minute time frame of EURUSD there was a 50.42% chance that the next candle would be bullish and 47.37% chance of a bearish candle and 2.21% chance open and close are equal

### Anaylyse candles after a prior large drop in prices up to 4.6 pips in a minute open_low

In [37]:
big_drop_hl_95 = dataset[dataset["high_low"] > 46]
big_drop_hl_95

open_close  open_low  open_high  close_low  close_high  high_low  up*  \
694          -35.0       0.0      -46.0       35.0       -11.0      46.0  1.0   
716          -67.0       0.0      -74.0       67.0        -7.0      74.0  1.0   
718          -47.0       0.0      -47.0       47.0         0.0      47.0  1.0   
764           47.0      52.0       -2.0        5.0       -49.0      54.0  1.0   
775           39.0      51.0       -4.0       12.0       -43.0      55.0  1.0   
...            ...       ...        ...        ...         ...       ...  ...   
371350       -36.0       1.0      -67.0       37.0       -31.0      68.0  1.0   
371352       -59.0       4.0      -63.0       63.0        -4.0      67.0  0.0   
371353        61.0      66.0      -10.0        5.0       -71.0      76.0  1.0   
371354       -52.0       0.0      -66.0       52.0       -14.0      66.0  1.0   
371357        49.0      53.0      -22.0        4.0       -71.0      75.0  1.0   

        down*  no_move*  
694       0.0       0.0  
716       0.0       0.0  
718       0.0       0.0  
764       0.0       0.0  
775       0.0       0.0  
...       ...       ...  
371350    0.0       0.0  
371352    1.0       0.0  
371353    0.0       0.0  
371354    0.0       0.0  
371357    0.0       0.0  

[19012 rows x 9 columns]

In [38]:
big_drop_hl_95[["up*", "down*", "no_move*"]].value_counts()

up*  down*  no_move*
1.0  0.0    0.0         9473
0.0  1.0    0.0         9172
     0.0    1.0          367
dtype: int64